### PR TITLE
Request types for filter management

### PR DESCRIPTION
**Pull request by Palakis**
- GetSourceFilters
- AddFilterToSource
- RemoveFilterFromSource
- SetSourceFilterSettings
- ReorderSourceFilter
- MoveSourceFilter

### CODE REVIEW

**Comment by Palakis** [comment]
@haganbmj @TStod @RytoEX May I ask one of you to have a look at this, please?

File: src/WSRequestHandler_Sources.cpp [hunk 1]
```diff
@@ -1313,3 +1313,314 @@ void WSRequestHandler::HandleGetSpecialSources(WSRequestHandler* req) {
...
+		req->SendErrorResponse("filter name already taken");
+		return;
+	}
+
+	OBSSourceAutoRelease filter = obs_source_create_private(filterType, filterName, filterSettings);
+	if (!filter) {
+		req->SendErrorResponse("filter creation failed");
+		return;
+	}
+	if (obs_source_get_type(filter) != OBS_SOURCE_TYPE_FILTER) {
```
Comment (haganbmj) [inline src/WSRequestHandler_Sources.cpp]: Out of curiosity, what other types could it be at this point?

**Comment by TStod** [comment]
IIRC OBS was investigating filters on a per scene item basis (instead of a per source basis) I'm unsure of the status of that change and will do some digging as it will inform our decisions.

File: src/WSRequestHandler_Sources.cpp (hunk 1 above)
Comment (Palakis) [reply src/WSRequestHandler_Sources.cpp]: @haganbmj Filters are sources, which type is one of the `OBS_SOURCE_TYPE_*` constants.

Comment (Palakis) [reply src/WSRequestHandler_Sources.cpp]: @haganbmj 
> what other types could it be at this point?

Actually I don't know. I just want to add another security to make sure the source created is indeed a filter.

Comment (haganbmj) [reply src/WSRequestHandler_Sources.cpp]: Yeah, I saw elsewhere in this commit where filters are `OBSSourceAutoRelease`, so makes sense to take precaution.

**Comment by RytoEX** [comment]
Should I look at anything in particular?

**Comment by Llorx** [comment]
How is this going right now?

**Comment by vicegold** [comment]
Any update on filter control via obs-websocket?

**Pull request merged**